src/components/ha-picker-combo-box.ts

```diff
@@ -159,94 +204,54 @@ export class HaPickerComboBox extends LitElement {
     return sortedItems;
   };
 
-  protected shouldUpdate(changedProps: PropertyValues) {
-    if (
-      changedProps.has("value") ||
-      changedProps.has("label") ||
-      changedProps.has("disabled")
-    ) {
-      return true;
-    }
-    return !(!changedProps.has("_opened") && this._opened);
-  }
-
-  public willUpdate(changedProps: PropertyValues) {
-    if (changedProps.has("_opened") && this._opened) {
-      this._items = this._getItems();
-      if (this._initialItems) {
-        this.comboBox.filteredItems = this._items;
-      }
-      this._initialItems = true;
-    }
-  }
+  private _renderItem = (item: PickerComboBoxItem, index: number) => {
+    const renderer = this.rowRenderer || DEFAULT_ROW_RENDERER;
+    return html`<div
+      id=${`list-item-${index}`}
+      class="combo-box-row ${this._value === item.id ? "current-value" : ""}"
+      .value=${item.id}
+      .index=${index}
+      @click=${this._valueSelected}
+    >
+      ${renderer(item, index)}
```

Comment:
Can we check for `NO_MATCHING_ITEMS_FOUND_ID` here so we don't call rowRenderer for this one and we directly direct it here. 
And the special case in rowRenderer for `NO_MATCHING_ITEMS_FOUND_ID` can be removed in every file.

Reply:
works :tada: thats a nice and clean solution